connectors/http/http-base/src/main/java/io/camunda/connector/http/base/components/apache/CustomApacheHttpClient.java

```diff
@@ -0,0 +1,136 @@
+
+  public static final CustomApacheHttpClient DEFAULT =
+      new CustomApacheHttpClient(createHttpClientBuilder());
+
+  private static final Logger LOG = LoggerFactory.getLogger(CustomApacheHttpClient.class);
+
+  private final HttpClientBuilder httpClientBuilder;
+
+  private CustomApacheHttpClient(HttpClientBuilder httpClientBuilder) {
+    this.httpClientBuilder = httpClientBuilder;
+  }
+
+  /**
+   * Returns the default instance of {@link CustomApacheHttpClient}. The default instance is
+   * configured with a {@link HttpClientBuilder} that has a {@link
+   * PoolingHttpClientConnectionManager} with a maximum of {@link Integer#MAX_VALUE} connections.
+   *
+   * @return the default instance of {@link CustomApacheHttpClient}
+   */
+  public static CustomApacheHttpClient getDefault() {
+    return DEFAULT;
+  }
+
+  /**
+   * Creates a new instance of {@link CustomApacheHttpClient} with the given {@link
+   * HttpClientBuilder}. Use this method if you want to customize the {@link HttpClientBuilder}. See
+   * {@link #getDefault()} for the default instance.
+   *
+   * @param httpClientBuilder the {@link HttpClientBuilder} to use
+   * @return a new instance of {@link CustomApacheHttpClient}
+   * @see HttpClients#custom()
+   */
+  public static CustomApacheHttpClient create(HttpClientBuilder httpClientBuilder) {
+    return new CustomApacheHttpClient(httpClientBuilder);
+  }
+
+  private static HttpClientBuilder createHttpClientBuilder() {
+    return HttpClients.custom()
+        .setConnectionManager(createConnectionManager())
+        .disableRedirectHandling();
+  }
+
+  private static PoolingHttpClientConnectionManager createConnectionManager() {
+    PoolingHttpClientConnectionManager connectionManager = new PoolingHttpClientConnectionManager();
+    connectionManager.setMaxTotal(Integer.MAX_VALUE);
+    connectionManager.setDefaultMaxPerRoute(Integer.MAX_VALUE);
+
+    // Socket config
+    connectionManager.setDefaultSocketConfig(SocketConfig.custom().setSoKeepAlive(true).build());
+
+    return connectionManager;
+  }
+
+  /**
+   * Converts the given {@link HttpCommonRequest} to an Apache {@link
+   * org.apache.hc.core5.http.ClassicHttpRequest} and executes it.
+   *
+   * @param request the request to execute
+   * @param remoteExecutionEnabled whether to use the internal Google Function to execute the
+   *     request remotely
+   * @return the {@link HttpCommonResult}
+   */
+  @Override
+  public HttpCommonResult execute(HttpCommonRequest request, boolean remoteExecutionEnabled)
+      throws Exception {
+    var apacheRequest = ApacheRequestFactory.get().createHttpRequest(request);
+    try {
+      HttpCommonResult response =
+          httpClientBuilder
+              .setDefaultRequestConfig(getRequestConfig(request))
+              .build()
+              .execute(apacheRequest, new HttpCommonResultResponseHandler(remoteExecutionEnabled));
+
+      if (response.getStatus() >= 400) {
```

Comment:
Thinking out loud: if we don't follow redirects, do we also consider a 3XX status an error? Maybe we could also throw an exception in this case to be consistent. WDYT?

Reply:
I don't have a strong opinion, in my mind a redirect is not an error + I wanted to keep the current behavior (I think the current Google HTTP client considers HTTP codes >4xx as errors).